README.md

```diff
@@ -219,6 +219,73 @@ nvim -u NORC -c "source https://raw.githubusercontent.com/nvim-neorocks/rocks.nv
 >
 > `LUA_BINDIR="${XDG_BIN_DIR:-$HOME/.local/bin}" LUA_BINDIR_SET=yes nvim -u NORC -c "source ...`
 
+### :rocket: Bootstrapping Script
+
+For those who want `rocks.nvim` to automatically install itself whenever it isn't installed
+one may use the bootstrapping script. Place the following script into your `init.lua`:
+
+<details>
+<summary>Lua Script</summary>
+
+```lua
+do
+    -- Specifies where to install/use rocks.nvim
+    local install_location = vim.fs.joinpath(vim.fn.stdpath("data"), "rocks")
+
+    -- Set up configuration options related to rocks.nvim (recommended to leave as default)
+    local rocks_config = {
+        rocks_path = vim.fs.normalize(install_location),
+        luarocks_binary = vim.fs.joinpath(install_location, "bin", "luarocks"),
+    }
+
+    vim.g.rocks_nvim = rocks_config
+
+    -- Configure the package path (so that plugin code can be found)
+    local luarocks_path = {
+        vim.fs.joinpath(rocks_config.rocks_path, "share", "lua", "5.1", "?.lua"),
+        vim.fs.joinpath(rocks_config.rocks_path, "share", "lua", "5.1", "?", "init.lua"),
+    }
+    package.path = package.path .. ";" .. table.concat(luarocks_path, ";")
+
+    -- Configure the C path (so that e.g. tree-sitter parsers can be found)
+    local luarocks_cpath = {
+        vim.fs.joinpath(rocks_config.rocks_path, "lib", "lua", "5.1", "?.so"),
+        vim.fs.joinpath(rocks_config.rocks_path, "lib64", "lua", "5.1", "?.so"),
+    }
+    package.cpath = package.cpath .. ";" .. table.concat(luarocks_cpath, ";")
+
+    -- Load all installed plugins, including rocks.nvim itself
+    vim.opt.runtimepath:append(vim.fs.joinpath(rocks_config.rocks_path, "lib", "luarocks", "rocks-5.1", "rocks.nvim", "*"))
+end
+
+-- If rocks.nvim is not installed then install it!
+if not pcall(require, "rocks") then
+    local rocks_location = vim.fs.joinpath(vim.fn.stdpath("cache"), "rocks.nvim")
+
+    if not vim.uv.fs_stat(rocks_location) then
+        -- Pull down rocks.nvim
+        vim.fn.system({
+            "git",
+            "clone",
+            "--filter=blob:none",
+            "https://github.com/nvim-neorocks/rocks.nvim",
+            rocks_location,
+        })
+    end
+
+    -- If the clone was successful then source the bootstrapping script
+    assert(vim.v.shell_error == 0, "rocks.nvim installation failed. Try exiting and re-entering Neovim!")
```

Comment:
nitpick (non-blocking): perhaps we could use `vim.system` API instead as we are already using Neovim nightly APIs?

Reply:
I chose to keep vim.fn.system just because vim.v.shell_error is nicer to deal with in this specific scenario. With vim.system we'd have to capture the output into a variable which would have to have a greater scope than the if statement -- too much haslle really :p